Bird indicators chapter

ChloeRN opened the pull request:
Submitting the draft of the bird indicators chapter. 

This needs reviewing not just by @anders-kolstad or @DrMattG, but also by @DiegoPavonJordan to ensure that the scripts are doing what they should.  


Like for my other sub-project, I have developed most of the code in a separate repository. Can be found here: https://github.com/NINAnor/birdIndicators

ChloeRN commented:
@anders-kolstad can you confirm that you are serious about wanting the final results in a wide format table, i.e. v_YYYY, sd_YYYY etc. should be **COLUMN NAMES** ?
I have implemented it that way but it just goes against everything I believe in 😜

ChloeRN commented:
@anders-kolstad and @DrMattG please also double-check the situation with the references. I don't think it's supposed to work when rendering single chapters, but please make sure it turns out correctly when rendering the book.

anders-kolstad reviewed (changes requested):
See comment about missing data object

anders-kolstad commented:
Thank you @ChloeRN . I started looking at it now, but got stuck due to a missing file.

Also about the wide format. You could be right. The idea was not to repeat the geometry column in the sf data. In any case you are the only one actually making it to the export stage so far, so I think I will just hash the part of the code that exports the data and also remove any output files from the repo, and then think about output formating later. It should probably go on OSF anyway, and not to GitHub.

ChloeRN commented:
@anders-kolstad which data object is missing?

anders-kolstad commented on DRAFT_birdIndicators.Rmd:
````diff
@@ -0,0 +1,481 @@
+The species list files should have been downloaded together with the rest of the `birdIndicators` repository and we can use `makeSpeciesLists()` to assemble the information. 
+
+```{r}
+sppLists <- makeSpeciesLists(Trim_data = Trim_data, spp_DataPath = "temp/birdIndicators-main/data")
````
I cant find _Trim_data_ which is created on line 220 with eval=F but not stored anywhere that I can see.

anders-kolstad replied:
There, I added the review again. Not sure what happened...

anders-kolstad reviewed (commented):
I'm unable to review before I can locate the Trim data

anders-kolstad commented:
Removing [eval=F](https://github.com/NINAnor/ecosystemCondition/pull/159/commits/35390937dd104f1fec70e2ec46be8fc3d5715e9b) made new error apprear:

```
Error: nanodbc/nanodbc.cpp:1135: 01S00: [Microsoft][ODBC SQL Server Driver][SQL Server]Cannot open database "TOVTaksering" requested by the login. The login failed.  [Microsoft][ODBC SQL Server Driver]Invalid connection string attribute 
```

This arrises in `DBI::dbConnect()` under `downloadData_TRIM()`

anders-kolstad commented:
@ChloeRN , I tried again to run `downloadData_TRIM()` from RStudio Server and now I get his error

```
Error: nanodbc/nanodbc.cpp:1125: 00000: [FreeTDS][SQL Server]Login failed for user 'NINA\anders.kolstad'.  [FreeTDS][SQL Server]Cannot open database "TOVTaksering" requested by the login. The login failed.  [FreeTDS][SQL Server]Unable to connect to data source 
```

My used does not have access is seems.

ChloeRN commented:
I think that means you need to contact Roald Vang and ask for access permission.
Otherwise:
I only use the downloaded data to make the "sppLists" object in the bookdown. I could instead add and read in this file as a .rds in the script. Would that be better?

________________________________
From: Anders L. Kolstad ***@***.***>
Sent: Tuesday, October 24, 2023 3:36 PM
To: NINAnor/ecosystemCondition ***@***.***>
Cc: Chloé R. Nater ***@***.***>; Mention ***@***.***>
Subject: Re: [NINAnor/ecosystemCondition] Bird indicators chapter (PR #159)


@ChloeRN<https://github.com/ChloeRN> , I tried again to run downloadData_TRIM() from RStudio Server and now I get his error

Error: nanodbc/nanodbc.cpp:1125: 00000: [FreeTDS][SQL Server]Login failed for user 'NINA\anders.kolstad'.  [FreeTDS][SQL Server]Cannot open database "TOVTaksering" requested by the login. The login failed.  [FreeTDS][SQL Server]Unable to connect to data source


My used does not have access is seems.

—
Reply to this email directly, view it on GitHub<https://github.com/NINAnor/ecosystemCondition/pull/159#issuecomment-1777221118>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJGP7ABX2A6LU3LPQPKWINTYA677LAVCNFSM6AAAAAA6KDC5D6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZXGIZDCMJRHA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

ChloeRN commented:
Will do this at the start of next week.
________________________________
From: Anders L. Kolstad ***@***.***>
Sent: Friday, October 27, 2023 10:16 AM
To: NINAnor/ecosystemCondition ***@***.***>
Cc: Chloé R. Nater ***@***.***>; Mention ***@***.***>
Subject: Re: [NINAnor/ecosystemCondition] Bird indicators chapter (PR #159)


Yes, perhaps cache the output and bypass this whole issue. And add a note as well that you need read persmissions to the data base in order to run the code?

—
Reply to this email directly, view it on GitHub<https://github.com/NINAnor/ecosystemCondition/pull/159#issuecomment-1782495961>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJGP7AD4Y7UK7CISVN2ZJMTYBNUXXAVCNFSM6AAAAAA6KDC5D6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTOOBSGQ4TKOJWGE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

ChloeRN commented:
@anders-kolstad I have now added the object `sppLists` as data and changed over to reading that in from folder instead of downloading the entire data from "Hekkefuglovervåkingen" to make it. 
Hopefully that will make the chapter compile for you.
